components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/cache/SessionContextCacheEntry.java

```diff
@@ -66,11 +69,21 @@ public void setAccessedTime() {
         this.accessedTime = System.currentTimeMillis();
     }
 
-    private void setAccessedTime(long accessedTime) {
+    public void setAccessedTime(long accessedTime) {
         this.accessedTime = accessedTime;
     }
 
     public long getAccessedTime() {
         return this.accessedTime;
     }
+
+    public OptimizedSessionContext getOptimizedSessionContext() {
+
+        return optimizedSessionContext;
+    }
+
+    public void setOptimizedSessionContext(OptimizedSessionContext optimizedSessionContext) {
```

Comment:
can we make the package private, so that it won't be exposed outside? same for getOptimizedSessionContext
```suggestion
    void setOptimizedSessionContext(OptimizedSessionContext optimizedSessionContext) {
```